test/recovery/recovery.c

```diff
@@ -0,0 +1,235 @@
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <sys/wait.h>
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
+#include <wiredtiger.h>
+
+#include "test_util.i"
+
+static char home[512];			/* Program working dir */
+static const char *progname;		/* Program name */
+static const char *uri = "table:main";
+
+#define	RECORDS_FILE "records"
+
+#define	ENV_CONFIG						\
+    "create,log=(file_max=10M,archive=false,enabled),"		\
+    "transaction_sync=(enabled,method=none)"
+#define	ENV_CONFIG_REC "log=(recover=on)"
+#define	MAX_VAL	4096
+
+static void
+usage(void)
+{
+	fprintf(stderr, "usage: %s [-h dir]\n", progname);
+	exit(EXIT_FAILURE);
+}
+
+/*
+ * Child process creates the database and table, and then writes data into
+ * the table until it is killed by the parent.
+ */
+static void
+fill_db()
+{
+	WT_CONNECTION *conn;
+	WT_CURSOR *cursor;
+	WT_ITEM data;
+	WT_RAND_STATE rnd;
+	WT_SESSION *session;
+	uint64_t i;
+	int fd, ret;
+	uint8_t buf[MAX_VAL];
+
+	__wt_random_init(&rnd);
+	memset(buf, 0, sizeof(buf));
+	/*
+	 * Initialize the first 25% to random values.  Leave a bunch of data
+	 * space at the end to emphasize zero data.
+	 */
+	for (i = 0; i < MAX_VAL/4; i++)
+		buf[i] = (uint8_t)__wt_random(&rnd);
+
+	/*
+	 * Run in the home directory so that the records file is in there too.
+	 */
+	chdir(home);
+	if ((ret = wiredtiger_open(NULL, NULL, ENV_CONFIG, &conn)) != 0)
+		testutil_die(ret, "wiredtiger_open");
+	if ((ret = conn->open_session(conn, NULL, NULL, &session)) != 0)
+		testutil_die(ret, "WT_CONNECTION:open_session");
+	if ((ret = session->create(session,
+	    uri, "key_format=Q,value_format=u")) != 0)
+		testutil_die(ret, "WT_SESSION.create: %s", uri);
+	if ((ret =
+	    session->open_cursor(session, uri, NULL, NULL, &cursor)) != 0)
+		testutil_die(ret, "WT_SESSION.open_cursor: %s", uri);
+
+	/*
+	 * Keep a separate file with the records we wrote for checking.
+	 */
+	unlink(RECORDS_FILE);
+
+	if ((fd = open(RECORDS_FILE, O_CREAT | O_WRONLY, 0777)) == -1)
```

Comment:
Should we open this file unbuffered? Is it possible that content will be lost when the child is killed?